src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -79,8 +79,10 @@ uct_cuda_ipc_iface_is_reachable_v2(const uct_iface_h tl_iface,
                                    const uct_iface_is_reachable_params_t *params)
 {
     return uct_iface_is_reachable_params_addrs_valid(params) &&
+#if !HAVE_CUDA_FABRIC
```

Comment:
how to properly check multi-node reachability?

Reply:
The correct way to check for reachability is to use nvml and see if clique, domain ids of GPUs match but iface_reachability is not a good place for this as we allow the flexibility of setting device context after worker creation but before transfer of buffers created on the specific cuda device context. For now, unconditionally packing the relevant export handle for the given buffer and failing during unpack (if not reachable) suffices. However, it does require unpack to fail gracefully and have UCX protov2 switch the next best protocol (like IB or host/device pipeline, etc)